src/LeafletField.php

```diff
@@ -86,146 +106,156 @@ public function Field($properties = array())
 
     /**
-     * {@inheritdoc}
+     * Return the L.map options.
+     *
+     * @return array
      */
-    public function saveInto(DataObjectInterface $record)
+    public function getMapOptions()
     {
-        if ($this->name) {
-            $record->setCastedField($this->name, $this->geometryField->dataValue());
-        }
+        return $this->options['map'];
     }
 
     /**
-     * @return FieldList
+     * Return the draw options as a json string.
+     *
+     * @return string
      */
-    public function getChildFields()
+    public function getDrawOptionsJS()
     {
-        return $this->children;
+        return Convert::array2json($this->getDrawOptions());
     }
 
     /**
-     * @return string
+     * Return the L.Control.Draw options.
+     *
+     * @return array
      */
-    public function getGeometry()
+    public function getDrawOptions()
     {
-        $fieldName = $this->getName();
-        return $this->data->$fieldName;
+        return $this->options['draw'];
     }
 
     /**
-     * Return the L.map options.
-     * @return Array
+     * Return additional geoJsonlayers to display on the map.
+     *
+     * @return string
      */
-    public function getMapOptions()
+    public function getGeoJsonlayersJS()
     {
-        return $this->options['map'];
+        return Convert::array2json($this->geoJsonlayers);
     }
 
     /**
-     * Return the map options as a json string.
+     * Return additional geoJsonlayers style to display on the map.
+     *
      * @return String
      */
-    public function getMapOptionsJS()
+    public function getGeoJsonlayersStyleJS()
     {
-        return Convert::array2json($this->getMapOptions());
+        return Convert::array2json($this->geoJsonlayersStyle);
     }
 
-    /**
-     * Set the map options, will override the config defaults.
-     * @param array $options
-     */
-    public function setMapOptions($options = array())
+    protected function requireDependencies()
     {
-        $this->options['map'] = array_merge($this->options['map'], $options);
+        Requirements::javascript('//cdnjs.cloudflare.com/ajax/libs/leaflet/1.0.3/leaflet.js');
+        Requirements::javascript('//cdnjs.cloudflare.com/ajax/libs/leaflet.draw/0.4.9/leaflet.draw.js');
+        Requirements::javascript('//maps.googleapis.com/maps/api/js?key=' . Environment::getEnv('GOOGLE_MAP_API_KEY'));
+        Requirements::javascript('//unpkg.com/leaflet.gridlayer.googlemutant@latest/Leaflet.GoogleMutant.js');
+        Requirements::javascript('nzta/googlemap-leafletfield: client/javascript/LeafletField.js');
+
+        Requirements::css('//cdnjs.cloudflare.com/ajax/libs/leaflet/1.0.3/leaflet.css');
+        Requirements::css('//cdnjs.cloudflare.com/ajax/libs/leaflet.draw/0.4.9/leaflet.draw.css');
+        Requirements::css('nzta/googlemap-leafletfield: client/css/LeafletField.css');
     }
 
     /**
-     * Return the L.Control.Draw options.
-     * @return Array
+     * {@inheritdoc}
```

Comment:
Dont think this should say `@inheritdoc`?

Reply:
No idea what is it and how it comes. change back to array